packages/babel-plugin-transform-regenerator/src/regenerator/leap.ts

```diff
@@ -1,176 +1,159 @@
 import assert from "node:assert";
 import { Emitter } from "./emit.ts";
-import { inherits } from "node:util";
 import { getTypes } from "./util.ts";
+import { types as t } from "@babel/core";
 
-function Entry(this: any) {
-  assert.ok(this instanceof Entry);
-}
-
-export function FunctionEntry(this: any, returnLoc: any) {
-  Entry.call(this);
-  getTypes().assertLiteral(returnLoc);
-  this.returnLoc = returnLoc;
-}
-
-inherits(FunctionEntry, Entry);
-
-export function LoopEntry(
-  this: any,
-  breakLoc: any,
-  continueLoc: any,
-  label: any,
-) {
-  Entry.call(this);
-
-  const t = getTypes();
-
-  t.assertLiteral(breakLoc);
-  t.assertLiteral(continueLoc);
-
-  if (label) {
-    t.assertIdentifier(label);
-  } else {
-    label = null;
+export class Entry {
+  constructor() {
+    assert.ok(this instanceof Entry);
   }
-
-  this.breakLoc = breakLoc;
-  this.continueLoc = continueLoc;
-  this.label = label;
 }
 
-inherits(LoopEntry, Entry);
-
-export function SwitchEntry(this: any, breakLoc: any) {
-  Entry.call(this);
-  getTypes().assertLiteral(breakLoc);
-  this.breakLoc = breakLoc;
+export class FunctionEntry extends Entry {
+  declare returnLoc: t.NumericLiteral;
+  constructor(returnLoc: t.NumericLiteral) {
+    super();
+    t.assertLiteral(returnLoc);
```

Comment:
We can remove all the asserts that now have equivalent (or better) type annotations.